tree/ntuple/src/RFieldMeta.cxx

```diff
@@ -330,7 +330,7 @@ ROOT::DescriptorId_t ROOT::RClassField::LookupMember(const ROOT::RNTupleDescript
 void ROOT::RClassField::SetStagingClass(const std::string &className, unsigned int classVersion)
 {
    TClass::GetClass(className.c_str())->GetStreamerInfo(classVersion);
-   if (classVersion != GetTypeVersion()) {
+   if (classVersion != GetTypeVersion() || className != GetTypeName()) {
```

Comment:
As a side note, comparing the TClass pointers is a tad bit more stable than comparing the classname as `TClass::Get Class` will deal with typographical and typedefs differences.

Reply:
Noted. In this case, we only have one `TClass` pointer (the other one is being figured out by the method) and we compare normalized type names.